test/sign.test.ts

```diff
@@ -1,12 +1,58 @@
+import { joinSignature } from "@ethersproject/bytes";
+import { hashMessage } from "@ethersproject/hash";
+import { SigningKey } from "@ethersproject/signing-key";
 import { expect } from "chai";
 import { ethers, waffle } from "hardhat";
 
 import {
   SigningScheme,
   signOrderCancellation,
   hashOrderCancellation,
+  signOrder,
 } from "../src/ts";
 
+import { SAMPLE_ORDER } from "./testHelpers";
+
+const patchedSignMessageBuilder = (key: SigningKey) => async (
+  message: string,
+): Promise<string> => {
+  // Reproducing `@ethersproject/wallet/src.ts/index.ts` sign message bahaviour
+  const sig = joinSignature(key.signDigest(hashMessage(message)));
+
+  // Unpack the signature
+  const { r, s, v } = ethers.utils.splitSignature(sig);
+  // Pack it again
+  return ethers.utils.solidityPack(
+    ["bytes32", "bytes32", "uint8"],
+    // Remove last byte's `27` padding
+    [r, s, v - 27],
+  );
+};
+
+describe("signOrder", () => {
+  it("should pad the `v` byte when needed", async () => {
+    const [signer] = waffle.provider.getWallets();
+    // Patch signMessage
+    signer.signMessage = patchedSignMessageBuilder(signer._signingKey());
```

Comment:
Here the assumption is that `signer.signMessage` resets to the original value for the following tests. This appears to be the case, but maybe it's worth anyway to restore it to the original value at the end of the test?

Reply:
That's a valid point. I have no idea.
Will do as you suggest.